src/dali/abstract_pair_converter_plugin.py

```diff
@@ -87,3 +117,109 @@ def get_conversion_rate(self, timestamp: datetime, from_asset: str, to_asset: st
             )
 
         return result
+
+    def _build_fiat_list(self) -> None:
+        try:
+            response: Response = self.__session.get(_EXCHANGE_SYMBOLS, timeout=self.__TIMEOUT)
+            # {
+            #     'motd':
+            #         {
+            #             'msg': 'If you or your company ...',
+            #             'url': 'https://exchangerate.host/#/donate'
+            #         },
+            #     'success': True,
+            #     'symbols':
+            #         {
+            #             'AED':
+            #                 {
+            #                     'description': 'United Arab Emirates Dirham',
+            #                     'code': 'AED'
+            #                 },
+            #             ...
+            #         }
+            # }
+            data: Any = response.json()
+            if data[_SUCCESS]:
+                self.__fiat_list = [fiat_iso for fiat_iso in data[_SYMBOLS] if fiat_iso != "BTC"]
+            else:
+                if "message" in data:
+                    LOGGER.error("Error %d: %s: %s", response.status_code, _EXCHANGE_SYMBOLS, data["message"])
+                response.raise_for_status()
+
+        except JSONDecodeError as exc:
+            LOGGER.debug("Fetching of fiat symbols failed. The server might be down. Please try again later.")
+            raise Exception("JSON decode error") from exc
+
+    def _add_fiat_graph_to(self, graph: Dict[str, List[str]], markets: Dict[str, List[str]]) -> None:
+        if not self.__fiat_list:
+            self._build_fiat_list()
+
+        for fiat in self.__fiat_list:
+            to_fiat_list: List[str] = self.__fiat_list.copy()
+            to_fiat_list.remove(fiat)
+            if graph.get(fiat):
+                for to_be_added_fiat in to_fiat_list:
+                    # add a pair if it doesn't exist
+                    if to_be_added_fiat not in graph[fiat]:
+                        graph[fiat].append(to_be_added_fiat)
+            else:
+                graph[fiat] = to_fiat_list
+
+            for to_fiat in to_fiat_list:
+                fiat_market = fiat + to_fiat
+                markets[fiat_market] = [_FIAT_EXCHANGE]
+
+            # Add prioritized fiat at the beginning
+            for priority_fiat in reversed(_FIAT_PRIORITY):
+                if priority_fiat in graph[fiat]:
+                    graph[fiat].insert(0, graph[fiat].pop(graph[fiat].index(priority_fiat)))
```

Comment:
This line also looks like it could benefit from switching from List to Set or Dict.